src/targets/gpu/fuse_mlir.cpp

```diff
@@ -591,6 +623,9 @@ void fuse_mlir::apply(module_pass_manager& mpm) const
         mpm,
         find_mlir_standalone_convolution_op{get_mode("convolution", mlir_mode::fast)},
         find_mlir_standalone_dot_op{get_mode("dot", mlir_mode::fast)});
+
+    if(enabled(MIGRAPHX_ENABLE_MLIR_INPUT_FUSION{}))
+        match::find_matches(mpm, find_pointwise_mlir{});
```

Comment:
1. At a minimum, we need to be making sure all the input ops are `is_pointwise_op_supported_by_mlir`
2. More generally, I'd ask that we have a separate filter function for what pointwise functions can be fused on the input, thus allowing us to filter off input fusions that aren't working

Reply:
3. The pointwise operation could be separated by a chain of views (reshape/transpose/...) - we should fuse those in too!
4. ... heck, we should do the same thing for non convolution/dot arguments (ex. the bias tensor), which we currently don't do

Reply:
> The pointwise operation could be separated by a chain of views (reshape/transpose/...) - we should fuse those in too!

We already fuse those on the inputs. This will fuse additionally on top of that.

Reply:
Added a seperate function for input fusion support.

Reply:
> "pull in the views" logic lives up in fuse_pointwise. 

I don't think we do this at the moment. @pfultz2  looks like it is something we can do on top of fusing pointwise inputs

Reply:
I mean currently it would fuse `pointwise + reshapes + gemm_based_ins` . We can do `reshapes + pointwise + reshapes + gemm_based_ins.`

Reply:
> I don't think we do this at the moment. @pfultz2 looks like it is something we can do on top of fusing pointwise inputs

Yes, we should open an issue to handle these cases.

Reply:
> I mean currently it would fuse pointwise + reshapes + gemm_based_ins . We can do reshapes + pointwise + reshapes + gemm_based_ins.

We could also handle `gemm_based_ins + reshapes + pointwise`, which we do see such patterns in models.

Reply:
#3211  Opened an issue.